### PR TITLE
Fixes bug with /src versions ARXIVCE-1455

### PR DESCRIPTION
**Pull request by bdc34**
Fixes bug where /src would return same source file for all versions of a paper.

### CODE REVIEW

**Review by bmaltzan — APPROVED** [review]

File: browse/services/dissemination/source_store.py [hunk 1]
```diff
@@ -50,16 +50,13 @@ def source_exists(self,
...
     def get_src(self, arxiv_id: Identifier, is_current: bool) -> Optional[FileObj]:
         if is_current:
-            parent = abs_path_current_parent(arxiv_id)
+            pattern = abs_path_current_parent(arxiv_id) + '/' + arxiv_id.id
         else:
             if not arxiv_id.has_version:
                 raise ValueError(f"arxiv_id {arxiv_id} does not have version")
             else:
-                parent = abs_path_orig_parent(arxiv_id)
+                pattern = abs_path_orig_parent(arxiv_id) + '/' + arxiv_id.idv
```
[the remark this answers is not in the document]
Comment (bmaltzan) [reply browse/services/dissemination/source_store.py:58]: Looks like idv already includes arxiv_id:
identifier/__init__.py:180:            self.idv = f'{self.id}v{self.version}'

**Comment by bmaltzan** [comment]
Maybe there's an extra "-" from a join somewhere.
Right before cond-mat:
FAILED tests/test_src.py::test_src[/src/-cond-mat/9805021v1-cond-mat9805021v1.ps.gz-gzipped ps] - assert 404 == 200

**Comment by bdc34** [comment]
> Maybe there's an extra "-" from a join somewhere. Right before cond-mat: FAILED tests/test_src.py::test_src[/src/-cond-mat/9805021v1-cond-mat9805021v1.ps.gz-gzipped ps] - assert 404 == 200

I'll take a look.

**Comment by bdc34** [comment]
> Maybe there's an extra "-" from a join somewhere. Right before cond-mat: FAILED tests/test_src.py::test_src[/src/-cond-mat/9805021v1-cond-mat9805021v1.ps.gz-gzipped ps] - assert 404 == 200

I fixed a regression detected by this test failure. :tada:

**Review by bmaltzan — APPROVED** [review]

**Pull request merged**